### ISSUE TITLE
Please do not file issues on Moonlight Windows until there is a version for the public

### ISSUE THREAD

**Issue by mrb113** [moonlight-stream/moonlight-windows#7]
Moonlight Windows is under development and is considered unusable; if you do install it onto your Windows device, please do not file any issues yet. 
We appreciate the interest in improving Moonlight Windows and will provide updates on the readme regarding development.

**Comment by TheEngineerGuy** [comment]
Hi Michelle,
Just going based on the note on homepage, isn't the low latency property available for Media Foundation on Windows 8 apps enough for this purpose? 

CODECAPI_AVLowLatencyMode property:
"This property applies to both encoders and decoders. 

Low-latency mode is useful for real-time communications or live capture, when latency should be minimized. However, low-latency mode might also reduce the decoding or encoding quality."

Ref: https://msdn.microsoft.com/en-us/library/windows/desktop/hh447590(v=vs.85).aspx

Considering that Windows has around 90% market share, this would certainly help the masses.

Please advise.

Thanks.

**Comment by mrb113** [comment]
In our investigations, low latency mode (achieved in Windows Universal apps by setting MediaElement's RealTimePlayback property to true) has not been sufficient for Moonlight. We suspect that there are a few bugs in Media Foundation regarding how it handles the specific format of our raw h264 stream. 

We have revisited the relevant APIs after each release of the Windows SDK and the issues still remain, most recently after the Threshold 2 update that rolled out in November. 

In short, we're able to stream, but the stream crashes often and the quality in general is not what we consider playable and fit for release.

We are considering reaching out to Microsoft in the future if we remain unable to work around these issues.

**Comment by TheEngineerGuy** [comment]
Thanks for a prompt response Michelle. That is quite sad to hear, when Microsoft has been peddling its WMF API for so many years, and has been touting the API capabilities of W10, still has bugs with H264 realtime on both.

Switching gears, (possibly not the right place to ask this question, but here it goes) is Java Client Hardware acceleration coming soon?

Please advise.

Thanks

**Comment by mrb113** [comment]
@cgutman that question is all you

**Comment by TheEngineerGuy** [comment]
Thanks for trying to help Michelle. I guess rather than fielding this question, Cameron is more comfortable removing the line about "Hardware acceleration" from the Java version's description, after the question was asked.

Anyway, I wish that we had more people like you in the dev community.

**Comment by cgutman** [comment]
I don't recall doing that. Anyway, there is no work being done on hardware acceleration in the Java client. I think the Java client should be replaced by a modified Moonlight-Embedded client. We didn't know it at the time, but using Java was a much bigger limitation than expected. Our best shot is probably a Qt GUI with the SDL backend of Moonlight-embedded. That or picking up the Kodi plugin developed for GSoC 2015.

I don't have time to do everything that needs doing. I'm currently the only person maintaining moonlight-common-c, moonlight-common (Java), moonlight-android, and moonlight-pc. The PC port usually gets the short end of the stick. I want to take a look at the Kodi plugin again around late January if time permits. Sharing Kodi's HW accelerated decoders could be a huge boon for maintainability, especially if it's still going to be just me doing it.

**Comment by rembomy** [comment]
Is this ever going to be reviewed?

**Comment by rabiet** [comment]
Well, I will look into this project when I find the time and try to help in some way or another. However I do not have that much experience in UWP programming and especially not networking or video decoding, so I doubt I will be much of help!

**Comment by Joe-Speers** [comment]
Anything new?

**Comment by TheEngineerGuy** [comment]
Possibly not, even though this feature is working between xbox and PC via UWP, and PS4 and PC via Win64.
